src/rstar.jl

```diff
@@ -1,15 +1,15 @@
 """
     rstar(
-        rng=Random.GLOBAL_RNG,
-        classifier,
-        samples::AbstractMatrix,
-        chain_indices::AbstractVector{Int};
+        rng::Random.AbstractRNG=Random.GLOBAL_RNG,
```

Comment:
> But maybe that's not supported by Julia 1.0? In any case, AFAIK Random.GLOBAL_RNG only exists for backwards compatibility.

Ah good to know! The docstrings all say the default is `Random.GLOBAL_RNG`, but indeed, since 1.3.0 the implementations have all used `default_rng()`. I'm not opposed to dropping support for <1.6, though this would only require us to drop <1.3.

Reply:
If Julia 1.3 still works, IMO there's no reason to drop support for it :slightly_smiling_face: I wonder, would that allow us to remove the Compat dependency again?

Reply:
Yes, we only need Compat to support v1.0. Even v1.1 would let us drop it. Although maybe this isn't even an issue now, because doesn't our `rstar` implementation already require v1.3? With our current version bounds, we can't even install our package and MLJBase in the same environment on v1.2.

Reply:
Well, it really depends on what model you want to use in `rstar`. Installing MCMCDiagnosticTools and MLJBase in the same environment works fine with e.g. Julia 1.0. But I'm not sure if there are any compatible MLJ classifiers. In the tests we also use MLJXGBoostInterface and that does not work on Julia < 1.3 (and also not on 32bit). But possibly some other classifier would work.

In any case, I think it would be reasonable to drop support for Julia < 1.3. Even more so since it will be a breaking release.